soc/xtensa/intel_adsp/ace/include/pm_data.h

```diff
@@ -0,0 +1,73 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * Author: Tomasz Leman <tomasz.m.leman@intel.com>
+ */
+
+#ifndef __INTEL_PM_DATA_H__
+#define __INTEL_PM_DATA_H__
+
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/__assert.h>
+
+/**
+ * @brief Structure to represent Intel platform-specific power management data.
+ *
+ * This structure contains data used for customization of power state operations
+ * on Intel platforms.
+ */
+struct intel_adsp_ace_pm_data {
+	/**
+	 * @brief Reference counter for lock of clock switching operation.
+	 *
+	 * This atomic variable ensures thread-safe increments and decrements
+	 * of the lock count, which is critical for managing clock switching
+	 * in a multi-core environment.
+	 */
+	atomic_t clock_switch_lock_refcount;
+};
+
+/**
+ * @brief Increment the clock switch lock reference counter.
+ *
+ * This function atomically increments the reference counter for the clock switch lock. It is used
+ * to prevent clock switching operations that change the current DSP clock when a core becomes
+ * active. The function returns true if this is the first core to request the lock, indicating that
+ * the clock switching operation was previously not locked.
+ *
+ * @param data Pointer to the intel_adsp_ace_pm_data struct containing the lock reference counter.
+ * @return True if this is the first increment operation, false otherwise.
+ */
+static ALWAYS_INLINE bool increment_clock_switch_lock(struct intel_adsp_ace_pm_data *data)
```

Comment:
Another nitpick: these two functions want to be namespaced to intel_adsp_ (or equivalent), the SOC headers tend to be included very widely even when you don't want them to. But again I really think we want this in the core PM layer instead.